examples/GetObjectRetentionConfig.java

```diff
@@ -0,0 +1,114 @@
+          InsufficientDataException, InternalException, ErrorResponseException,
+          InvalidBucketNameException, InvalidPortException, InvalidEndpointException,
+          RegionConflictException, IllegalArgumentException {
+    try {
+
+      /* play.min.io for test and development. */
+      MinioClient minioClient =
+          new MinioClient(
+              "https://play.min.io",
+              "Q3AM3UQ867SPQQA43P2F",
+              "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG");
+
+      // Create bucket if it doesn't exist.
+      boolean found = minioClient.bucketExists("my-bucketname");
+      if (found) {
+        System.out.println("my-bucketname already exists");
+      } else {
+        // Create bucket 'my-bucketname' with object lock functionality enabled
+        minioClient.makeBucket(
+            MakeBucketArgs.builder().bucket("my-bucketname").objectLock(true).build());
+        System.out.println(
+            "my-bucketname is created successfully with object lock functionality enabled.");
+      }
+
+      StringBuilder builder = new StringBuilder();
+      for (int i = 0; i < 100; i++) {
+        builder.append(
+            "Sphinx of black quartz, judge my vow: Used by Adobe InDesign to display font samples. ");
+        builder.append("(29 letters)\n");
+        builder.append(
+            "Jackdaws love my big sphinx of quartz: Similarly, used by Windows XP for some fonts. ");
+        builder.append("---\n");
+      }
+
+      // Create a InputStream for object upload.
+      ByteArrayInputStream bais = new ByteArrayInputStream(builder.toString().getBytes("UTF-8"));
+
+      // Create object 'my-objectname' in 'my-bucketname' with content from the input stream.
+      minioClient.putObject(
+          "my-bucketname", "my-objectname", bais, new PutObjectOptions(bais.available(), -1));
+      bais.close();
+      System.out.println("my-objectname is uploaded successfully");
+
+      // Declaring config with Retention mode as Compliance and
+      // retain until one year to current date.
+      ZonedDateTime retentionUntil = ZonedDateTime.now().plusYears(1);
+      Retention config = new Retention(RetentionMode.COMPLIANCE, retentionUntil);
+
+      // Set object lock configuration
+      minioClient.setObjectRetention(
+          SetObjectRetentionArgs.builder()
+              .bucket("my-bucketname")
+              .object("my-objectname")
+              .config(config)
+              .bypassGovernanceRetention(true)
+              .build());
+
+      // Get object lock retention
+      Retention retention =
+          minioClient.getObjectRetention(
+              GetObjectRetentionArgs.builder()
+                  .bucket("my-bucketname")
+                  .object("my-objectname")
+                  .build());
+
+      System.out.println("Mode: " + retention.mode());
+      System.out.println("Retainuntil Date: " + retention.retainUntilDate());
```

Comment:
I defer by providing example from bucket creation to the end. My suggestion is to specific to the API.